django/db/models/indexes.py

```diff
@@ -60,16 +79,44 @@ def _get_condition_sql(self, model, schema_editor):
         sql, params = where.as_sql(compiler, schema_editor.connection)
         return sql % tuple(schema_editor.quote_value(p) for p in params)
 
+    def _get_ordered_field(self, model, field_name):
+        """
+        Return the field for supplied name and an optional ordering
+        """
+        field_name, order = (
+            (field_name[1:], 'DESC') if field_name.startswith('-') else (field_name, '')
+        )
+        field = model._meta.get_field(field_name)
+        return field, order
+
+    def _get_field_orders(self, model):
+        """
+        Return fields together with their ordering
+        """
+        fields = []
+        for field in self.fields:
+            fields.append(self._get_ordered_field(model, field))
+        return fields
+
```

Comment:
These hooks are not related to this PR, I think we can revert them or move to a separate commit/PR. @hannseman What do you think?

Reply:
I don't have a strong opinion here, they are legacy from when we passed compiled expressions in `_create_index_sql(fields=..)`.

Reply:
I think we should remove them.